btree/btree.c

```diff
@@ -5975,66 +5915,35 @@ static void ut_basic_tree_oper(void)
 	/**
 	 *  Run a valid scenario which:
 	 *  1) Creates a btree
-	 *  2) Closes the btree
-	 *  3) Opens the btree
-	 *  4) Closes the btree
-	 *  5) Destroys the btree
+	 *  2) Opens the btree
+	 *  3) Closes the btree
+	 *  4) Destroys the btree
 	 */
 
 	/** Create temp node space*/
 	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
 	btree = m0_alloc(sizeof *btree);
 	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_create(temp_node, 1024,
-							     &btree_type, nt,
-							     &b_op, seg, tx));
+				      &btree_type, nt, &b_op, seg, tx));
 	M0_ASSERT(rc == 0);
 
-	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_close(b_op.bo_arbor,
-							    &b_op));
-	M0_ASSERT(rc == 0);
-	temp_btree = b_op.bo_arbor;
 	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_open(temp_node, 1024,
 							   &btree, &b_op));
 	M0_ASSERT(rc == 0);
 
 	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_close(btree, &b_op));
 	M0_ASSERT(rc == 0);
-	b_op.bo_arbor = temp_btree;
 
-	if (b_op.bo_arbor->t_desc->t_ref > 0) {
-		rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op,
-					      m0_btree_destroy(b_op.bo_arbor,
-							       &b_op, tx));
-		M0_ASSERT(rc == 0);
-	}
+	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_destroy(b_op.bo_arbor,
```

Comment:
In m0_btree_close the tree descriptor will get freed. After that t_desc is deferenced in m0_btree_destroy.